ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/activity-picklist-remote-auto-complete-options.component.ts

```diff
@@ -0,0 +1,133 @@
+import { ConfigurationService } from '../../../services/configuration.service';
+import { NGXTranslateService } from '../../../services/internationalization/ngxTranslate.service';
+import { PicklistSortingPolicy } from '../../../services/picklistSortingPolicy.service';
+import { BaseActivityPicklistQuestion } from './baseActivityPicklistQuestion.component';
+
+@Component({
+    selector: 'ddp-activity-picklist-remote-auto-complete-options',
+    template: `<mat-form-field
+        class="full-width ddp-autocomplete-field"
+        *ngIf="picklistOptions$ | async as picklistOptions"
+    >
+        <input
+            matInput
+            #autocompleteInput
+            [value]="searchValue$ | async"
+            [placeholder]="block.picklistLabel"
+            [matAutocomplete]="autoCompleteFromSource"
+            (keyup)="onInput($event.target.value)"
+        />
+
+        <mat-autocomplete
+            #autoCompleteFromSource="matAutocomplete"
+            class="autoCompletePanel"
+            [displayWith]="displayAutoComplete"
+        >
+            <mat-option
+                *ngFor="let option of picklistOptions"
+                class="autoCompleteOption"
+                [value]="option"
+                (click)="onValueSelect(option)"
+            >
+                <span
+                    [innerHtml]="
+                        option.optionLabel
+                            | searchHighlight
+                                : autocompleteInput.value
+                                : ignoredSymbolsInQuery
+                    "
+                ></span>
+            </mat-option>
+        </mat-autocomplete>
+    </mat-form-field>`,
+    styles: [
+        `
+            .full-width {
+                width: 100%;
+            }
+
+            .ddp-autocomplete-field ::ng-deep .mat-form-field-infix {
+                width: auto;
+            }
+        `,
+    ],
+})
+export class ActivityPicklistRemoteAutoCompleteOptionsComponent
+    extends BaseActivityPicklistQuestion
+    implements OnInit
+{
+    @Input() studyGuid: string;
+    @Input() activityGuid: string;
+    filteredGroups: ActivityPicklistNormalizedGroup[] = [];
+
+    searchValue$ = new BehaviorSubject('');
+    picklistOptions$: Observable<ActivityPicklistOption[]>;
+
+    readonly ignoredSymbolsInQuery: string[];
+
+    constructor(
+        translate: NGXTranslateService,
+        private sortPolicy: PicklistSortingPolicy,
+        @Inject('ddp.config') public config: ConfigurationService // private activityService: ActivityServiceAgent
+    ) {
+        super(translate);
+        this.ignoredSymbolsInQuery =
+            this.config.picklistAutocompleteIgnoredSymbols;
+    }
+
+    ngOnInit(): void {
+        console.log(this.studyGuid, this.activityGuid);
+        this.picklistOptions$ = this.searchValue$.pipe(
+            switchMap((searchValue) => {
+                this.sortPolicy = this.shouldBeSorted
```

Comment:
I guess it would be enough to define/init `this.sortPolicy` once (not for every searchValue change).

Reply:
Makes sense, refactoring